### ISSUE TITLE
Locked balance doesn't always get updated

### ISSUE THREAD

**Issue by a123b** [bisq-network/bisq#2305]
When you are the maker of a trade and your offer gets accepted, the reserved funds are locked in a multisig address. This is correctly reflected in the "Locked Funds" tab. In the summary at the top right of the app, however, this doesn't always seem to be the case. I noticed many times now that it shows "0.00 BTC locked" even though there are locked funds. It only gets updated to the actual locked value after restarting the app.

**Comment by ghost** [comment]
Thanks for the return !

**Comment by ripcurlx** [comment]
@a123b When you experience this problem does the reserved balance not update as well or is just the locked balance displayed wrong?

**Comment by ripcurlx** [comment]
When this issue is occuring, could you try (if you have another own offer available) to disable and enable the offer to see if the balance updates afterwards correctly? Changing the open offer list, should also trigger the update of all balances. I just want to make sure, that the problem lies in a listener not being triggered or if it's something else.

**Comment by ripcurlx** [comment]
Just from a code point of view the locked funds tab and the locked balance get their data from the same source. The only difference is, that the locked balance sums up the balance of failed, closed and locked trades. But as you didn't get an exception when you have this issue, I don't think there is an issue in the sum up code. Also just to rule out a general JavaFx rendering issue: when you experience this problem try to click somewhere else on your desktop and select afterwards the Bisq app again to force a change of focus, which triggers a redraw within the JavaFX render pipeline.

**Comment by a123b** [comment]
The reserved balance correctly updated to 0.00 when the trade was accepted. (That's why I noticed the bug, I saw that 0.01 BTC seemed to suddenly have disappeared...)
I don't think it was a rendering issue, I minimized, switched windows, moved the Bisq window quite a few times over a period of at least a day.

The summing code can't be the issue as the balance is correct after a restart. Unfortunately, I very rarely have multiple offers open at the same time, so I can't test if it gets updated after disabling/enabling offers.

If all balances are always updated at the same time, I could imagine that this is some kind of race condition: The update is triggered after the offer is accepted, so the balance isn't reserved anymore, but at this point the multisig doesn't show in the "Locked Funds" tab yet because the transaction isn't broadcast or confirmed yet.
Could it be something like this?
